### ISSUE TITLE
DatePicker: timeOnly moves focus to popup

### ISSUE THREAD

**Issue by jepsar** [primefaces/primefaces#13891]
### Describe the bug

When I tab to a `<p:datePicker timeOnly="true"/>`, the popup will open and the focus will be put in there. Instead the focus should stay in the input field, so you can enter the time. Now there is no way to get back into the input field and you are forced to use the popup UI.

Ok, I just figured out that "Esc" closes the popup. I think an extra tab from the input field would be a better option though. I don't think my users will know to use escape.

### Reproducer

https://www.primefaces.org/showcase/ui/input/datepicker/datePickerJava8.xhtml

### Expected behavior

Focus stays in the input field allowing you to type (faster).

### PrimeFaces edition

Community

### PrimeFaces version

15.0.3

### Theme

Freya

### JSF implementation

Mojarra

### JSF version

4

### Java version

21

### Browser(s)

Chrome macOS

**Comment by melloware** [comment]
Does showOnFocus=false work?

**Comment by jepsar** [comment]
When I combine it with showIcon="true", it works for me. But I still feel that the default focus behavior is at least weird.

**Comment by melloware** [comment]
Well `showOnFocus="false"` you should be able to press SPACE to open the panel and ESC to close the panel.  Else if your users really need mouse then `showIcon="true"` gives them a visual indicator or button to click on to bring up the panel.

I am open to suggestions but its always worked like this since PrimeTek introduced the component?

**Comment by melloware** [comment]
Actually looking at PrimeVue they used to focus the component on popup on entry but now they don't: https://primevue.org/datepicker/#accessibility

So that changed since I did the accessibility updates it always popped up and focused "Opens popup and moves focus to the selected date, if there is none focuses on today."

**Comment by jepsar** [comment]
In Vue it is working how I expected it to work.

**Comment by melloware** [comment]
Yep they changed that Vue used to work just like ours does now...that is what I modeled it after.

**Comment by melloware** [comment]
@jepsar my only concern with this is ARIA guidelines say if a popup is displayed it MUST have focus and should be "focus trapped" to that popup.  Because a visually impaired user won't know the popup is up if the InputText has focus. So I think PrimeVue is actually violating ARIA guidelines now by doing what its doing.

Found this: https://www.washington.edu/accesstech/checklist/popups/

**Comment by jepsar** [comment]
"Avoid triggering a popup on hover or focus. Instead, the user should specifically choose to expose the hidden content by clicking, tapping, or pressing enter or space. For more on this issue, see [Predictability](https://www.washington.edu/accesstech/checklist/predictability/)."

**Comment by melloware** [comment]
yep

**Comment by jepsar** [comment]
Isn't making the default of `showOnFocus="true"` enough here? Maybe combined with a default of `showIcon="true"`? It's a minor breaking (?) change which users can easily override globally.

**Comment by melloware** [comment]
yeah that might not be a bad idea.

**Comment by jepsar** [comment]
Is that something we can do within 15? I mean introduce a breaking change (albeit a small one).

**Comment by melloware** [comment]
@tandraschko thoughts?  

We could mark it breaking change and reason is "aligning with PrimeVue"

Actually @jepsar your also not the first person to ask for this so to me it feels like better default.

**Comment by melloware** [comment]
@jepsar since you can do this with Global Attributes I moved this to 16.0.0: https://primefaces.github.io/primefaces/15_0_0/#/core/globalattributes is that OK?
